File: DSA2026/twosum.ts
const target = 10;
const nums = [1, 2, 3, 4, 5, 6, 7, 8, 9];

function twoSum(nums: number[], target: number): number[] {
  let prevMap = new Map();
  for (let index = 0; index < nums.length; index++) {
    const element = nums[index];
    let diff = target - element;

    if (prevMap.has(diff)) {
      return [prevMap.get(diff), index];
    } else {
      prevMap.set(element, index);
    }
  }
  return [];
}

console.log(twoSum(nums, target));

File: DSA2026/twosum.py
nums = [2,6,1,8,9]
target = 14

def twosum(nums, target):
    prevMap = {}
    
    for i,n in enumerate(nums):
        diff = target-n
        if diff in prevMap:
            return [prevMap[diff], i ]
        prevMap[n] = i
    return

print(twosum(nums, target))